transformBAM.sh

```diff
@@ -51,37 +51,90 @@ while [ "$1" != "" ]; do
 done
 
 logMsg "INFO" "--------------------- START Transformation -----------------------"
-logMsg "INFO" "Creating new header"
 if [[ ! -e "$FILE_IN" ]];then
     logMsg "ERROR" "Input file does not exists: ($FILE_IN)"
 fi
-if [[ ! -e "$(dirname $FILE_IN)" ]];then
-    logMsg "WARN" "Output folder does not exists: ($FILE_IN).\nCreating it now."
-    mkdir -P $(dirname "$FILE_IN")
+DIR_IN="$(pwd)"
+logMsg "INFO" "Calculating number of entries in original BAM."
+originalSize=$(samtools view -c -@ 8 "$DIR_IN/$FILE_IN")
+if [[ ! -e "$DIR_OUT" ]];then
+    logMsg "WARN" "Output folder does not exists: ($DIR_OUT).\nCreating it now."
+    mkdir -p $DIR_OUT
 fi
-samtools view -H $FILE_IN -@ 8 | sed "s%$PM_IN%$PM_OUT%g" > BAM.new.header.sam.txt
-logMsg "INFO" "Checking new header for PM IDs"
-if [[ $(grep -c $PM_IN BAM.new.header.sam.txt) -gt 0 ]];then
-        logMsg "ERROR" "Found PM ID in header"
-fi
-logMsg "INFO" "Checking BAM file for PM IDs"
-if [[ $(samtools view $FILE_IN | grep -c $PM_IN) -lt 1 ]];then
-        logMsg "INFO" "No PM IDs found in BAM file. Simply re-head the BAM file"
-        samtools reheader -P BAM.new.header.sam.txt $FILE_IN > $FILE_OUT
+logMsg "DEBUG" "Directory Out: ($DIR_OUT)"
+FILE_OUT=$(echo $FILE_IN | sed "s%$PM_IN%$PM_OUT%g")
+logMsg "DEBUG" "File Name Out: ($FILE_OUT)"
+
+cd "$DIR_OUT"
+logMsg "INFO" "Making sure it's in samtools v10 format"
+samtools view -H "$DIR_IN/$FILE_IN" -@ 8 | sed "s%FCID:%FC:%g" | sed "s%BCID:%BC:%g" | sed "s%LNID:%LN:%g" > BAM.v10.header.txt || logMsg "ERROR" "Something wrong with generating the v10 header."
+samtools reheader -P BAM.v10.header.txt "$DIR_IN/$FILE_IN" > sample.v10.bam || logMsg "ERROR" "Something wrong with v10 header transformation."
+rm BAM.v10.header.txt
+logMsg "INFO" "Checking that the v10 version is equivalent to the original BAM"
+newSize=$(samtools view -c -@ 8 sample.v10.bam)
+if [[ ! $originalSize -eq $newSize ]];then
+        logMsg "ERROR" "The v10 version is different from the original BAM file."
```

Comment:
Is this a condition that should halt the execution or we can tolerate it?

Reply:
This is halt the execution. The "ERROR" as message type ensure that the execution is stopped.